.github/workflows/on-merge-group-and-push-branch-main.yml

```diff
@@ -1,7 +1,9 @@
-name: On Merge Group Branch "main"
+name: On Merge Group And Push Branch "main"
 
 on:
   merge_group:
+    types: [checks_requested]
+  push:
     branches:
```

Comment:
Removed `branches` from `merge_group`, re-added `push`, and added `branches` to `push`. 

When used with `merge_group`, `branches` appears to means something different than what it means for `push`. 

With `push`, is means when something is merged into `main`. With `merge_group` is seems to mean when `main` is merged into something else. Or at least that's my guess. Because this workflow didn't run when I merged https://github.com/CrowdStrike/glide-core/pull/775.